### DISCUSSION TITLE
Distributed SQL for even better performance/scaling?

### DISCUSSION THREAD

**Discussion by helpisdev** [geldata/gel#4720] [Ideas]
I don't know if it is currently doable/possible, but I believe that this awesome project would go to a whole other level, if it used distributed PostgreSQL under the hood. Distributed dbs have much to offer, and there are a lot out there, and for PostgreSQL the open source [Citus](https://www.citusdata.com/) extension does the job. It seems to extend more or less the same components of PostgreSQL that EdgeDB utilizes, and that got me thinking: "what if these two worked together?".

I'm really curious to see how other people feel about this so I would like to hear everyone's opinion on this, as it would probably be a huge architectural change.

**Comment by agritheory** [comment]
According to the test suite, the HA option in use is called Stolon, which I'm on my way to read about...
https://github.com/sorintlab/stolon
